flocker/node/agents/blockdevice.py

```diff
@@ -408,6 +419,34 @@ def run(self, deployer):
         return succeed(None)
 
 
+def _ensure_no_filesystem(device):
+    """
+    Raises an error if there's already a filesystem on ``device``.
+
+    :raises: ``FilesystemExists`` if there is already a filesystem on
+        ``device``.
+    :return: ``None``
+    """
+    try:
+        check_output(
+            [b"blkid", b"-p", b"-u", b"filesystem", device.path],
+            stderr=STDOUT,
+        )
+    except CalledProcessError as e:
+        # According to the man page:
+        #   the specified token was not found, or no (specified) devices
+        #   could be identified
+        #
+        # Experimentation shows that there is no output in the case of the
+        # former, and an error printed to stderr in the case of the
+        # latter.
+        if e.returncode == 2 and not e.output:
+            # There is no filesystem on this device.
+            return
+        raise
```

Comment:
Agreed, which is why I didn't test it :wink:

Have filed [FLOC-2388](https://clusterhq.atlassian.net/browse/FLOC-2388). Comments welcome.